### ISSUE TITLE
Raspberry Pi Zero unstable with 6.0.1

### ISSUE THREAD

**Issue by azbesthu** [OpenELEC/OpenELEC.tv#4670]
After a few test with Openelec 6.0.1 it seems the stock 1GHz speed with Raspi Zero is very unreliable.

The following issues occurring frekvently:
- expanding partition never ends
- After reboot only a "-" character shows up on screen
- After reboot / shutdown filesystem corruption detected

I used Strongtonics 2A power supply, but also tested with an other kind of 2A device. Voltage checked with Charger doctor device. No rainbow square or red led blinking. Also tested with powered usb hub for usb-lan and keyboard. I tried with Sandisk and Samsung sd cards. Those are working fine with other Pi devices (B1rev2, 2B...). So it is not the usual power issue :)

After these issues I tried to set the clock speeds to the same as the default for other Raspi1 devices:
arm_freq=700
core_freq=250
sdram_freq=400

Now the Zero is booting and working fine. No sign for sd card corruption and no more freezing. 
Not sure if config.txt has filter for zero (like [pi1] and [pi2]), so maybe it would be better to set a default "safe" clock in the next release for raspi1 devices. Maybe this factory overclock to 1GHz is too much for some of the Zeros.

The 6.0.0 release handled the Zero like any other pi1 devices. The firmware set the clock for 700MHz for the Zero, too. After the updater firmware in 6.0.1 it is now on 1GHz.

**Comment by popcornmix** [comment]
Can you determine which of the clocks are not stable?
Default is:
arm_freq=1000
core_freq=400
sdram_freq=450

Can you try increasing each in turn?
This is the first report I've seen of Pi0 not being stable at 1GHz.

**Comment by azbesthu** [comment]
Sure.

I'm doing the following: Starting it. Playing some tv stream with tvheadend pvr client, rebooting from the GUI. If everything seems to be ok, then I change config.txt over ssh and run reboot command. And repeate the steps with the new values.

- First I changed only  the sdram_freq to 450. It's ok. (temperature around 45C)
- Then only sdram_freq to 400. It's ok. (Temperature around 48C)
- Then only arm_freq to 1000. (Temperature around 50C) After some video playing, pressing reboot on the gui, Zero cannot boot because 
"*** Error in mount_storage: mount_common: Could not mount /dev/mmcblk0p2 ***
### Starting debugging shell ......

So setting arm_freq seems to be enought to reproduce the issue (1000/250/400). 
The room temperature is 22C, the Zero is in the air, so the desk does not cools the board.

**Comment by azbesthu** [comment]
Also 700/400/450 seems to be stable.

**Comment by MilhouseVH** [comment]
Apparently the default `over_voltage` for the Pi Zero is 6, are you using a different (lower) voltage setting (intentionally or by accident)? Does setting `over_voltage=6` or even `over_voltage=8` improve stability with `arm_freq=1000`?

**Comment by azbesthu** [comment]
overvoltage is unchanged, it is commented out by default:
"# over_voltage=0"

Now, I try the default settings (everything is commented out), but adding 
over_voltage=6, it failed to boot, on startup already filesystem corruption detected.

**Comment by popcornmix** [comment]
Note: Pi zero doesn't have under-voltage detection, so you wont see the coloured square or flashing LED. I suspect that low voltage is the most likely cause of the instability.
You need to measure the voltage on the Pi zero itself when under load (i.e. running kodi)

**Comment by azbesthu** [comment]
Right, I forgot that is only B+ and 2B related.

But, as I said in the first comment, I use charger doctor to monitor voltage. It is connected to the Zeros usb port. It says stable 5.13V-5.14V while booting and failing to mount. 
Amper is 0 because my powered usb hub has diode to prevent back-powering. So the connected usb devices uses power from the hubs power adapter, not from Zero.

**Comment by azbesthu** [comment]
So, I got in my hands an other Zero to try it with the same environment. The second one seems to be working fine.

That means this issue is invalid. The problem is not with Openelec, but with my actual device.

**Comment by chewitt** [comment]
Closing the ticket based on user feedback ^^ Thanks.

**Comment by edd3a** [comment]
I had instability issues with a Pi zero and the OTG ethernet network. It came up (or not) very unreliable. I replaced the usb power cable, and have [edited](http://openelec.tv/forum/124-raspberry-pi/79023-raspberry-pi-zero?start=30) my config.txt with:

arm_freq=700
core_freq=250
gpu_freq=250
sdram_freq=400

Looks like I am in business now. Initially I had also issues with a stable video out, which were solved by upgrading the power adapter.
